### PR TITLE
EmuFileWrapper: Properly memset the file structure

### PR DESCRIPTION
**Pull request by fritsch**
Do not change behaviour. Make sure all fields are properly set to zero.

### CODE REVIEW

**Comment by fritsch** [comment]
@Paxxi if it compiles it should restore old behavior.

File: xbmc/cores/DllLoader/exports/util/EmuFileWrapper.cpp [hunk 1]
```diff
@@ -73,17 +77,11 @@ void CEmuFileWrapper::CleanUp()
         delete m_files[i].file_lock;
         m_files[i].file_lock = nullptr;
       }
-#if !defined(TARGET_WINDOWS)
       //Don't memset on Windows as it overwrites our pointer
```
Comment (MrMC) [inline xbmc/cores/DllLoader/exports/util/EmuFileWrapper.cpp]: does this comment no longer apply ?

Comment (fritsch) [inline xbmc/cores/DllLoader/exports/util/EmuFileWrapper.cpp]: See what it does? It was not memset before, cause of the PlaceHolder ptr  should not be overwritten. No the PlaceHolder is deleted prior the memsetting. That's the idea here.

Comment (MrMC) [inline xbmc/cores/DllLoader/exports/util/EmuFileWrapper.cpp]: yes, I see what it does. 

the ifdef was removed, and the comment seems to apply to why the ifdef was there. So if you removed the ifdef, seems the comment should also get removed ? or moved up into new and changed to reflect why that bit of code is there.

Comment (fritsch) [inline xbmc/cores/DllLoader/exports/util/EmuFileWrapper.cpp]: The commend needs to go, yes.

**Comment by fritsch** [comment]
@MrMC thx for the comment - I think there is another problem in the Unregister method, which ommits the memset but only sets the use field to false ...  that cries like possible leak.

**Pull request closed without merging**